### ISSUE TITLE
Issues with libinput/dconf/sddm

### ISSUE THREAD

**Issue by tobiasBora** [NixOS/nixpkgs#293278]
### Describe the bug

I want `setxkbmap -option ctrl:swapcaps` to work (i.e. exchange my left CTRL and caps lock keys), I want to be able to configure some touchpad options (using libinput ideally) and I also want to be able to use the virtual keyboard when logging in SDDM. Unfortunately, it seems like I can’t find any configuration where this work. For instance, if I enable `libinput`, then `setxkbmap -option ctrl:swapcaps` does not work as intended (both keys are turned into caps lock). This is fixed if I use `programs.dconf.enable = true;` (seems very magic) but then in SDDM I cannot enable the virtual keyboard anymore.

### Steps To Reproduce

Add in the configuration:
```
  programs.dconf.enable = true;
  services.xserver.displayManager.sddm.enable = true;
  services.xserver.desktopManager.plasma5.enable = true;
  services.xserver.xkb.layout = "us";
  services.xserver.xkb.variant = "intl";
  services.xserver.libinput = {
    enable = true;
    touchpad = {
      # Very annoying dragLock: when you untouch the device, for a few ms you can still
      # continue the locking...
      tappingDragLock = false;
      disableWhileTyping = true;
      accelSpeed = "0.3";
    };
    mouse = {
      middleEmulation = true;
    };
  };

```

and try to enable the virtual keyboard in sddm. Disable dconf:  `setxkbmap -option ctrl:swapcaps`  stops working as expected (same if you configure it from KDE’s settings; but it seems like sometimes this one fails while setxkbmap works).

### Expected behavior

I would expect `swapcaps` and virtual keyboard to work irrespective of the setting I use. So maybe enable dconf by default and understand why the virtual keyboard stops working? See also the bug I filled in SDDM https://github.com/sddm/sddm/issues/1880 and https://github.com/sddm/sddm/issues/1879

Btw, shouldn’t dconf be automatically enabled by default in KDE since it seems needed for stuff like `ctrl:swapcaps`?

### Notify maintainers

@K900 sorry, you are the last editor of the libinput file and you also modified sddm.

### Metadata

Running NixOs 23.11, 79baff8812a0d68e24a836df0a364c678089e2c7

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
We do enable dconf by default in Plasma setups on unstable. How are you running setxkbmap?

**Comment by tobiasBora** [comment]
Oh really, weird that I see changes depending on whether I enable it or not… I just run it from Konsole like:
```
$ setxkbmap fr bepo
$ setxkbmap -option ctrl:swapcaps
```
See also the bug I filled in SDDM https://github.com/sddm/sddm/issues/1880

**Comment by K900** [comment]
But how do you expect that to affect SDDM, which runs a completely separate compositor/X server?

**Comment by tobiasBora** [comment]
Ok, sorry, so I did a bit more checks and tests, and:
1. dconf is enabled on unstable but not on 23.11
1. Indeed, you are right, SDDM is in fact a completely unrelated bug. In fact, it depends on which field you click. If you click directly on password, then it is fine, you have access to the virtual keyboard (actually, I discovered another bug: I don’t know why but I cannot directly click on it, I need first to go to TTY 1 and come back to TTY 2 to be able to do any click… no idea if this is a hardware bug on my side, or on SDDM or NixOs, but I can reproduce this quite accurately. Dell touchpads are really bad in my experience, this one keeps clicking mechanically for no reason so it might be a hardware bug, just confused why it is solved when switching to TTY…)
2. The bug with setxkbmap is in fact not really with setxkbmap, but rather with KDE itself. To be more precise, if I type:
```
$ setxkbmap fr bepo
$ setxkbmap -option ctrl:swapcaps
```
then everything works as expected (even the bug with shortcuts that I describe below is solved). If I type only:
```
$ setxkbmap -option ctrl:swapcaps
```
then, this command seems to do nothing at all, and keeps KDE’s setting (that’s why I was confused when I first filled this bug). But KDE’s settings are as follows:

![image](https://github.com/NixOS/nixpkgs/assets/2164118/56106510-2f4e-44da-90bb-a517b84ef23f)

![image](https://github.com/NixOS/nixpkgs/assets/2164118/cfda1ee3-01e2-4737-8b0b-377e38e5b03d)

So I would expect ctrl and swap to be exchanged… but it is not the case as both are caps lock. So this bug in fact can be drammatically cleanned up as: *KDE’s ctrl swap with caps lock is not applied correctly*. It might be simpler to close this bug and start a new one…
